TheForceEngine/TFE_FrontEndUI/frontEndUi.cpp

```diff
@@ -2231,6 +2234,37 @@ namespace TFE_FrontEndUI
 
 		TFE_Audio::setVolume(sound->soundFxVolume);
 		TFE_MidiPlayer::setVolume(sound->musicVolume);
+
+		int MIDI_DeviceCount = TFE_MidiDevice::getDeviceCount();
+		const char* MIDI_Devices;
+		std::string MIDI_DevicesStr;
+		
+		for (int i = 0; i < MIDI_DeviceCount; i++) {
+			MIDI_DevicesStr += TFE_MidiDevice::getDeviceNameStr(i) + '\0';
+		}
+		MIDI_Devices = &MIDI_DevicesStr[0];
+		
+		ImGui::Dummy(ImVec2(0.0f, 20.0f));
+
+		ImGui::LabelText("##ConfigLabel", "MIDI Device");
+		ImGui::LabelText("##ConfigLabel", "# of devices: %i", TFE_MidiDevice::getDeviceCount());
+		ImGui::SetNextItemWidth(512.0f);
+		static int MIDI_CurrentDevIndex = TFE_Settings::getSoundSettings()->midiDevice;
+
+
+		if (ImGui::Combo("##MIDI Device", &MIDI_CurrentDevIndex, MIDI_Devices, MIDI_DeviceCount)) {
+			sound->midiDevice = MIDI_CurrentDevIndex;
+			TFE_MidiPlayer::destroy();
+			TFE_MidiPlayer::init(MIDI_CurrentDevIndex);
+			MIDI_ShowDevChangeAlert = true;
```

Comment:
Could you please wrap this in
if (MIDI_CurrentDevIndex != TFE_Settings::getSoundSettings()->midiDevice) {
[...]
}
No need to do anything or show the warning if the selection hasn't changed.
Maybe even disable the combobox if only one entry is available?

Reply:
I went ahead and added this.
Thanks!